samples/core/resource_spec/runtime_resource_request_gpu.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import kfp
+from kfp import dsl, components
+from typing import NamedTuple
+
+
+def training_job():
+    import torch
+    use_cuda = torch.cuda.is_available()
+    print(f"The gpus status is: {use_cuda}")
+    if not use_cuda:
+        raise ValueError("GPU not available")
+
+
+training_comp = components.create_component_from_func(
+        training_job,
+        base_image='pytorch/pytorch:1.7.1-cuda11.0-cudnn8-runtime',
+        packages_to_install=[]
+    )
+
+@components.create_component_from_func
+def generate_resource_request() -> NamedTuple('output', [('gpu_vendor', str), ('nbr_gpus', str), ('constrain_type', str), ('constrain_value', str)]):
+    '''Returns the gpu resource settings'''
+    from collections import namedtuple
+    resource_output = namedtuple('output', ['gpu_vendor', 'nbr_gpu', 'constrain_type', 'constrain_value'])
```

Comment:
I was under the impression that the first argument to `namedtuple` needs to be the same as the left-hand side of the assignment. 
Maybe that's not a must-have but a convention?

Reply:
It is not a requirement, https://docs.python.org/3/library/collections.html#collections.namedtuple. Example: 

```
from collections import namedtuple
Point = namedtuple('Tmp', ['x', 'y'])
p = Point(11, y=22)
p[0] + p[1]
```

>Maybe that's not a must-have but a convention?

Yes, it seems to be the convention, will update it.

Reply:
Found the following from the doc you referenced.

> To support pickling, the named tuple class should be assigned to a variable that matches typename.

So it's bit more than just a convention.  :)

Reply:
> So it's bit more than just a convention. :)

Good catch!